### ISSUE TITLE
General information - SF-SAC data entry

### ISSUE THREAD

**Issue by MelissaBraxton** [GSA-TTS/FAC#185]
**Background**

This story covers the front end and back end implementation of the **general information page** of SF-SAC form data entry. We have mocked this and tested our mocks in sprint 1 research, but as of 5/5, refinements are in progress [here](https://www.figma.com/file/1bNncVt1TcFFDhQslqFC2o/FAC-Playground-(Copy)?node-id=2464%3A10748).  You can find where this sites in the story map [here](https://app.mural.co/t/gsa6/m/gsa6/1647625636085/7825fa5ead63ebb1b65eefab5ea20d24a8ab7c59?wid=0-1647626671985).

It covers the following stories:

1.  As an auditor or auditee, I can enter or edit general info and have it populate my SF-SAC form so that I can complete my audit.
2. As an auditor or auditee, I can have basic information about my entity prepopulated from sam.gov so that I can save time
3. As an auditor or auditee, audit period years shown on the general info page are carried over from what I input when I created the audit to create a new audit so that I can be sure I'm working on the correct version of the SF-SAC form without reentering data.
4.  As an auditor or auditee inputting submission data, I would like to be able to navigate through the different form sections from the bottom of each page.

**Acceptance criteria (We'll know we're done when...)**
from [implementation notes](https://www.figma.com/file/1bNncVt1TcFFDhQslqFC2o/FAC-Playground-(Copy)?node-id=2464%3A10748)
- [ ] The form is fully functional and all stories can be tested on the deployment server
- [ ] Fiscal period start and end dates should be autopopulated from the auditee info page. These fields should not be editable here. 
- [ ] Auditee UEI should be autopopulated from the auditee info page. This field should not be editable here. 
- [ ] Auditee name - should be autopopulated from the auditee info page. This field should not be editable here. 
- [ ] Auditee address should be autopopulated from SAM.gov. This field should not be editable here. 
- [ ] Auditee email - Should allow folks to select from emails they entered on the submission access page.
- [ ] Primary auditor email - Should allow folks to select from emails they entered on the submission access page.
- [ ] All required fields show validation messages
- [ ] Handling the special case of manually-entered data that would normally be provided by SAM.gov via UEI lookup

Move to another ticket:
- [ ] Multiple EINs - By clicking “yes,” here this unlocks the “multiple EINs page of the SF-SAC. Clicking “no” will mean that section remains locked.
- [ ] Multiple UEIs - By clicking “yes,” here this unlocks the “multiple UEIs page of the SF-SAC. Clicking “no” will mean that section remains locked.

```[tasklist]
- [x] Design refines mocks based on sprint 1 research
- [x] https://github.com/GSA-TTS/FAC/issues/216
- [ ] https://github.com/GSA-TTS/FAC/issues/284
  - [ ] https://github.com/GSA-TTS/FAC/issues/398
  - [ ] https://github.com/GSA-TTS/FAC/issues/399
- [x] https://github.com/GSA-TTS/FAC/issues/217
- [ ] https://github.com/GSA-TTS/FAC/issues/402
- [ ] #818
- [ ] #819
- [ ] https://github.com/GSA-TTS/FAC/issues/218
```

**Review checklist:**
- [ ] Meets acceptance criteria
- [ ] Screen reader - Listen to the experience with a screen reader extension, ensure the information presented in order
- [ ] Keyboard navigation - Run through acceptance criteria with keyboard tabs, ensure it works. 
- [ ] Text scaling - Adjust viewport to 1280 pixels wide and zoom to 200%, ensure everything renders as expected. Document 400% zoom issues with USWDS if appropriate.

**Comment by MelissaBraxton** [comment]
Without that story, we will need to ask users to re-enter audit period dates. What will we do if those dates they enter on the general info page do not match the dates we validate on that auditee info page? 

That story is about carrying those dates forward and preventing duplicate data entry and possible discrepancies between audit period dates if users are asked to manually enter those dates at multiple points in the process. 

Is it not feasible to auto-populate those validated dates throughout the form?

**Comment by petermarks-gsa** [comment]
@MelissaBraxton Oh, I think I misunderstood the story. I re-read and now I understand it's talking about carrying over data from a previous form in the same year, not a different audit year. I'll undo the strike through. Sorry for the confusion on my end.

**Comment by MelissaBraxton** [comment]
@sreuber We shouldn't need dummy data for this task--it's about mocking the form in its empty state. 

To move story 2 forward, we need to finalize **which fields** on this page will be auto-populated by SAM.gov. We mocked auto-populating the fields that SAM.gov has available in what we tested in Sprint 1, but we should propose which fields we think should be auto-populated moving forward, and then invite engineering to weigh in. Probably doing that in a comment on this issue will suffice.

**Comment by MelissaBraxton** [comment]
FYI, @sreuber - with this note, I went ahead and checked off the first task. 

Expected save behavior: 

**Record (but don't save) changes as they're made, but then prompt users to save or discard any unsaved changes if they try to navigate away from the page.**

We will cover the design and implementation of the prompt to save or discard unsaved date in a new issue.

**Comment by MelissaBraxton** [comment]
Taking @sreuber off since the design work is now done. Other remaining work is covered in other stories that have other folks assigned.

**Comment by petermarks-gsa** [comment]
@jasnakai @geekygirlsarah I understand the two of you met about your remaining tasks on this story. Can I assign the story itself to one of you to "shepherd" this through the QA process? By "shepherd", I mean making sure what's deployed on the staging server meets acceptance criteria and is ready to be QA'ed by me. Thanks!

**Comment by heymatthenry** [comment]
After discussing it, @timoballard & I think that #284 (and its respective front- & back-end breakout issues) effectively captures the scope for the milestone 2 iteration of this ticket. So that brackets out both the "Next Iteration" bullets from the issue description, as well as the Multiple UEIs/EINs page bullets, which have already been ID'd as candidates for separate tickets.

**Comment by petermarks-gsa** [comment]
@heymatthenry @timoballard Following up on estimation meeting today, flagging that the goal is to estimate this before sprint planning tomorrow. Re-adding needs scoping label until that's done.

**Comment by heymatthenry** [comment]
@timoballard Given the number of breakout tickets, and the FE/BE coordination this will take, this feels `XL` to me. What do you think?

**Comment by timoballard** [comment]
@heymatthenry agree, definitely `XL`

**Comment by petermarks-gsa** [comment]
Sounds good. I'll remove "needs scoping" as I believe that was the only thing outstanding here.

**Comment by tadhg-ohiggins** [comment]
Moving to blocked on the basis that until #399 and #391 are done nobody is working on this.

**Comment by pkigongo** [comment]
RE: 818, 819 - 
Focused on UEI; When someone enters UEI, we're supposed to check that  in SAM.gov; Let the user agree that this is who they're submitting for...
